### PR TITLE
[WIP][agnhost] Use a specific mirror instead of CDN for alpine

### PR DESCRIPTION
**Pull request by dims**
Looks like everyone is doing this :(
https://grep.app/search?q=sed.%2Adl-cdn.alpinelinux.org&regexp=true

The retry we added [earlier](https://github.com/kubernetes/kubernetes/pull/101800) helped a bit, but not fully:
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-e2e-agnhost-test-images/1390766452872581120

More info on the clarkson mirror is here:
https://mirror.clarkson.edu/distributions.html

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/images/agnhost/Dockerfile [hunk 1]
```diff
@@ -17,6 +17,10 @@ FROM $BASEIMAGE
 
 CROSS_BUILD_COPY qemu-QEMUARCH-static /usr/bin/
 
+# Switch to Apline Linux Mirror of Clarkson University, the mirror system is incredibly flaky:
+# https://mirror.clarkson.edu/distributions.html#alpine
+RUN sed -i 's/http\:\/\/dl-cdn.alpinelinux.org/http\:\/\/mirror.clarkson.edu/g' /etc/apk/repositories
```
Comment (aojea) [inline test/images/agnhost/Dockerfile:22]: unrelated to the PR but curious, how hard is to set up our own mirror in k8s infra?

**Comment by BenTheElder** [comment]
why this particular mirror?

**Comment by dims** [comment]
@BenTheElder was just trying something out :) filing another PR now

**Pull request closed without merging**